agent/agent.go

```diff
@@ -1814,6 +1815,20 @@ func (a *Agent) RemoveService(serviceID string, persist bool) error {
 	}
 
 	a.logger.Printf("[DEBUG] agent: removed service %q", serviceID)
+
+	// If any Sidecar services exist for the removed service ID, remove them too.
+	if sidecar := a.State.Service(a.sidecarServiceID(serviceID)); sidecar != nil {
```

Comment:
SidecareServiceID is `serviceID + "-sidecar-proxy"` - how can there be a collision? I am assuming that the serviceids are generated and have a specific format.

Reply:
I learned that service ids can be provided by the user which means that collisions are possible.

nevermind.